src/utils/env.ts

```diff
@@ -14,12 +14,11 @@ export const getRequiredProcessEnv = (name: string): string => {
   return envVar;
 };
 
-export function getOptionalProcessEnv(name: string, defaultVar?: never): null;
-
+export function getOptionalProcessEnv<T>(name: string): null | string;
 export function getOptionalProcessEnv<T>(name: string, defaultVar: T): T | string;
 
 // eslint-disable-next-line prefer-arrow/prefer-arrow-functions
-export function getOptionalProcessEnv<T>(name: string, defaultVar?: T): null | T | string {
+export function getOptionalProcessEnv<T>(name: string, defaultVar?: T | undefined): (null | T) | string {
   return hasProcessEnv(name) ? getRequiredProcessEnv(name) : defaultVar ?? null;
 }
```

Comment:
what do you think about returning this?


```ts
export function getOptionalProcessEnv(name: string, defaultVar?: never): null;

export function getOptionalProcessEnv(name: string, defaultVar: unknown): string;

// eslint-disable-next-line prefer-arrow/prefer-arrow-functions
export function getOptionalProcessEnv(name: string, defaultVar?: unknown): null | string {
  return hasProcessEnv(name) ? getRequiredProcessEnv(name) : defaultVar == null ? null : String(defaultVar);
}
```

Reply:
that could work but i'm a bit worried by changing the `defaultVar`.

idk if we'd ever do this but calling `String(x)` could cause issues with an env var that's JSON:

```js
// if doing String(defaultVar)
const SOME_VAR = JSON.parse(getOptionalProcessEnv('SOME_VAR', JSON.stringify({ a: 1 }))); 
```

```js
// if returning unmodified
const SOME_VAR = JSON.parse(getOptionalProcessEnv('SOME_VAR', { a: 1 })); 
```
is this a reasonable risk?

Reply:
Don't think we ever will use json in vars, but even if so, I still think the default values should be a string, or at least values which can be converted to string. Otherwise, we will have the same problems as in your second example(parse function can't parse default value, since it's object, it means we should add some checks, which is increasing code complexity).

I guess we can implement something like this, it should work fine for json values too, what do you think?:

```typescript
export function getOptionalProcessEnv(name: string, defaultVar?: unknown): null | string {
  if (hasProcessEnv(name)) {
    return getRequiredProcessEnv(name);
  }

  if (defaultVar !== null && typeof defaultVar === 'object') {
    return JSON. stringify(defaultVar);
  }
  
  return String(defaultVar)
}
```

Reply:
yeah that's a good point. i updated the PR to reflect your suggestions